stixcore/products/quicklook.py

```diff
@@ -0,0 +1,376 @@
+        return cls(control, data)
+
+    @classmethod
+    def from_fits(cls, fitspath):
+        header = fits.getheader(fitspath)
+        header_info = {'service_type': int(header.get('stype')),
+                       'service_subtype': int(header.get('sstype')),
+                       'pi1_val': int(header.get('ssid')),
+                       'spid': int(header.get('spid'))}
+        control = QTable.read(fitspath, hdu='CONTROL')
+        data = QTable.read(fitspath, hdu='DATA')
+        obs_beg = DateTime.from_string(header['DATE_OBS'])
+        data['time'] = (data['time'] + obs_beg.as_float())
+        return cls(**header_info, control=control, data=data)
+
+    def get_energies(self):
+        if 'energy_bin_edge_mask' in self.control.colnames:
+            energies = get_energies_from_mask(self.control['energy_bin_edge_mask'][0])
+        elif 'energy_bin_mask' in self.control.colnames:
+            energies = get_energies_from_mask(self.control['energy_bin_mask'][0])
+        else:
+            energies = get_energies_from_mask()
+
+        return energies
+
+
+class LightCurve(Product):
+    """
+    Quick Look Light Curve data product.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.name = 'LightCurve'
+        self.level = 'L0'
+
+    @classmethod
+    def from_packets(cls, packets):
+        service_type = packets.get('service_type')[0]
+        service_subtype = packets.get('service_subtype')[0]
+        spid = packets.get('spid')[0]
+        pi1_val = packets.get('pi1_val')[0]
+
+        control = Control.from_packets(packets)
+        control['detector_mask'] = _get_detector_mask(packets)
+        control['pixel_mask'] = _get_pixel_mask(packets)
+        control['energy_bin_edge_mask'] = _get_energy_bins(packets, 'NIX00266', 'NIXD0107')
+        control['compression_scheme_counts_skm'] = \
+            _get_compression_scheme(packets, 'NIXD0101', 'NIXD0102', 'NIXD0103')
+        control['compression_scheme_triggers_skm'] = \
+            _get_compression_scheme(packets, 'NIXD0104', 'NIXD0105', 'NIXD0106')
+        control['num_energies'] = _get_num_energies(packets)
+        control['num_samples'] = np.array(packets.get('NIX00271')).flatten()[
+            np.cumsum(control['num_energies']) - 1]
+
+        time, duration = control._get_time()
+        # Map a given entry back to the control info through index
+        control_indices = np.hstack([np.full(ns, cind) for ns, cind in
+                                     control[['num_samples', 'index']]])
+
+        cs, ck, cm = control['compression_scheme_counts_skm'][0]
+        counts = np.hstack(packets.get('NIX00272'))
+        counts, counts_var = decompress(counts, s=cs, k=ck, m=cm, return_variance=True)
+
+        ts, tk, tm = control['compression_scheme_triggers_skm'][0]
+        triggers = np.hstack(packets.get('NIX00274'))
+        triggers, triggers_var = decompress(triggers, s=ts, k=tk, m=tm, return_variance=True)
```

Comment:
i still think we should use our generic: `stixcore.processing.decompression.decompress(packet)`

this will keep the code much cleaner as all hard coded NIXNames are abstracted hidden internally.